### PR TITLE
Http

### PR DESCRIPTION
**Pull request by robertdrygala**
I used dynamodb + http server on aws for services.

### CODE REVIEW

File: src/app/user/auth.service.ts [hunk 1]
```diff
@@ -1,31 +1,62 @@
...
+  constructor(@Inject(APP_STORAGE) private storage: Storage,private http: HttpClient) {}
 
   login(user: any, password: any) {
     console.log('Login user : ' + user);
     this.storage.setItem('user', user);
     this.storage.setItem('password', password);
+    this.http.get<String>(environment.angular_course_api_gateway_auth).pipe(
+      tap(_ => this.log('fetched courses')),
+      catchError(this.handleError<String>('login')),
+    ).toPromise().then(value => {
```
Comment (SergioV85) [inline src/app/user/auth.service.ts]: Why you convert Observable to Promise?

File: src/app/course-details/course-details.component.ts [hunk 2]
```diff
@@ -7,12 +7,42 @@ import { CourseServiceService } from '../services/course-service.service';
...
+    this.courseService.filterCourses(this.titleFilter).subscribe(courseWrapper => {
+      console.log('Fetched data : ' + courseWrapper.Count);
+      this.courseItems = courseWrapper.Items;
+    });
+  }
+
+  public removeItem(item: CourseItem) {
+
+    const index = this.courseItems.indexOf(item, 0);
+    if (index > -1) {
```
[the remark this answers is not in the document]
Comment (robertdrygala) [reply src/app/course-details/course-details.component.ts]: Ok

File: src/app/pipes/orderby.pipe.ts [hunk 3]
```diff
@@ -7,13 +7,15 @@ import { CourseItem } from '../model/course-item';
 export class OrderByPipe implements PipeTransform {
   transform(allCurseItems: CourseItem[]) {
     allCurseItems.sort((item1, item2) => {
-      if (item1.creationDate.getTime() > item2.creationDate.getTime()) {
-        return 1;
-      }
+      // TypeError: condition.getTime is not a function FIX it somehow
```
[the remark this answers is not in the document]
Comment (robertdrygala) [reply src/app/pipes/orderby.pipe.ts]: Done

File: src/app/directives/plate-border.directive.ts [hunk 4]
```diff
@@ -13,17 +13,18 @@ export class PlateBorderHiglightDirective {
       console.warn('Condition date not set!');
       return;
     }
+    // TypeError: condition.getTime is not a function FIX it somehow
```
[the remark this answers is not in the document]
Comment (robertdrygala) [reply src/app/directives/plate-border.directive.ts]: Done

File: src/app/user/auth.service.ts (hunk 1 above)
Comment (robertdrygala) [reply src/app/user/auth.service.ts]: I don't remember :)

**Pull request merged**